CODING_GUIDELINES.md

```diff
@@ -6,30 +6,86 @@ This document contains the coding guidelines for the ASPIRE Python project.
-2. GitHub guides from [GitHub](https://guides.github.com/) 
+2. GitHub guides from [GitHub](https://guides.github.com/)
 
-## Git Workflow Changes
+## Git Workflow
 
-Our project follows [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
+Our project follows a reduced [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
 Please submit any PRs against the `develop` branch.
 
 ![Gitflow Diagram](https://wac-cdn.atlassian.com/dam/jcr:61ccc620-5249-4338-be66-94d563f2843c/05%20(2).svg?cdnVersion=357)
 
+The basic idea is that we have `feature` branches,  `develop` as an integration focused branch, and `master` for releases.
+We do not currently have a need for a dedicated release or staging branch; in our case `develop` also serves this purpose.
+Generally feature branches should branch off of the latest `develop` branch.
+
+Developers are welcome to maintain their own fork, but this is not required.
+
+When `develop` is at a milestone where we want to cut a release we will commit with a PEP440 compliant `Major.Minor` version
+scheme, tag it similarly, and complete a merge to master following the controls below. It is at this point the
+release would be a candadite to upload to PyPI.
+
+###  Git Workflow Controls
+
+To facilitate effective integration, merges into `develop` must occur as a pull request that is approved by two reviewers.
+One reviewer can be anyone, but should probably be someone related to the effort if that is applicable.
+The second developer is a designated codeowner described in the `CODEOWNERS` file.
+
+Additionally, all outstanding requested changes and comments must be resolved before merging.  Pushing new changes should trigger
+a re-review process of changed files to prevent oversights.
+
+Code should be passing integration and unit tests as defined in code on all supported plaforms by the evolving continuous integration systems,
+or be explicitly exempted by a codeowner. This is both for code quality and so maintenence of CI systems is not overlooked.
+
+All of these controls will be managed automatically by the Github server.
+
+(discussion note, maybe want an additional control (Amit?) on master/releases, I don't expect too often..).
+
+### Git Workflow Courtesy
+
+Contributors are responsible for keeping their working repositories up to date with the main Github repo.
+This is particularly important during active development.
+
+If a piece of work is known to change files that will effect other developers' active work, they should be invited to the review or tagged,
+so that the most effective order of integration can be established.  In the case of your own works having integration
+concerns, again, inviting other developers to discuss might be helpful.
+
+In the case a codeowner will not be available, such as vacation/travel, and an emergency patch can not be delayed
```

Comment:
“In case a codeowner is not available (such as sickness, vacation, travel, and so on) and an emergency patch cannot be delayed, a simple consensus…”

Reply:
Happy to change the language, but I was more wondering are we really planning on having something that can't be delayed into master?.. Maybe its not needed.

Reply:
I mean, it's a rare situation, but if it occurs, it's good to have a basis for making these type of consensus decisions.

Reply:
Sure, np, I pasted the change in there.

Reply:
Thanks.